pkg/testsuite/mock/wallet_transactions.go

```diff
@@ -0,0 +1,386 @@
+func (w *Wallet) CreateImplicitAccountFromInput(transactionName string, inputName string, recipientWallet *Wallet) *iotago.SignedTransaction {
+	input := w.Output(inputName)
+
+	implicitAccountOutput := &iotago.BasicOutput{
+		Amount: MinIssuerAccountAmount,
+		Mana:   AccountConversionManaCost,
+		Conditions: iotago.BasicOutputUnlockConditions{
+			&iotago.AddressUnlockCondition{Address: recipientWallet.ImplicitAccountCreationAddress()},
+		},
+		Features: iotago.BasicOutputFeatures{},
+	}
+
+	remainderBasicOutput := &iotago.BasicOutput{
+		Amount: input.BaseTokenAmount() - MinIssuerAccountAmount,
+		Mana:   input.StoredMana() - AccountConversionManaCost,
+		Conditions: iotago.BasicOutputUnlockConditions{
+			&iotago.AddressUnlockCondition{Address: input.Output().UnlockConditionSet().Address().Address},
+		},
+		Features: iotago.BasicOutputFeatures{},
+	}
+
+	signedTransaction := lo.PanicOnErr(w.createSignedTransactionWithOptions(
+		WithInputs(utxoledger.Outputs{input}),
+		WithOutputs(iotago.Outputs[iotago.Output]{implicitAccountOutput, remainderBasicOutput}),
+	))
+
+	// register the outputs in the wallet
+	w.registerOutputs(transactionName, signedTransaction.Transaction)
+	recipientWallet.registerOutputs(transactionName, signedTransaction.Transaction)
+
+	// register the implicit account as a block issuer in the wallet
+	implicitAccountID := iotago.AccountIDFromOutputID(recipientWallet.Output(fmt.Sprintf("%s:0", transactionName)).OutputID())
+	recipientWallet.AddBlockIssuer(implicitAccountID)
+
+	return signedTransaction
+}
+
+func (w *Wallet) TransitionImplicitAccountToAccountOutput(transactioName string, inputName string, creationSlot iotago.SlotIndex, opts ...options.Option[builder.AccountOutputBuilder]) *iotago.SignedTransaction {
+	input := w.Output(inputName)
+	implicitAccountID := iotago.AccountIDFromOutputID(input.OutputID())
+
+	basicOutput, isBasic := input.Output().(*iotago.BasicOutput)
+	if !isBasic {
+		panic(fmt.Sprintf("output with alias %s is not *iotago.BasicOutput", inputName))
+	}
+	if basicOutput.UnlockConditionSet().Address().Address.Type() != iotago.AddressImplicitAccountCreation {
+		panic(fmt.Sprintf("output with alias %s is not an implicit account", inputName))
+	}
+
+	accountOutput := options.Apply(builder.NewAccountOutputBuilder(w.Address(), w.Address(), MinIssuerAccountAmount).
+		AccountID(iotago.AccountIDFromOutputID(input.OutputID())),
+		opts).MustBuild()
+
+	signedTransaction := lo.PanicOnErr(w.createSignedTransactionWithAllotmentAndOptions(
+		creationSlot,
+		implicitAccountID,
+		WithContextInputs(iotago.TxEssenceContextInputs{
+			&iotago.BlockIssuanceCreditInput{
+				AccountID: implicitAccountID,
+			},
+			&iotago.CommitmentInput{
+				CommitmentID: w.Node.Protocol.MainEngineInstance().Storage.Settings().LatestCommitment().Commitment().MustID(),
+			},
+		}),
+		WithInputs(utxoledger.Outputs{input}),
+		WithOutputs(iotago.Outputs[iotago.Output]{accountOutput}),
+		WithSlotCreated(creationSlot),
+	))
```

Comment:
```suggestion
	signedTransaction := lo.PanicOnErr(w.createSignedTransactionWithOptions(
		WithContextInputs(iotago.TxEssenceContextInputs{
			&iotago.BlockIssuanceCreditInput{
				AccountID: implicitAccountID,
			},
			&iotago.CommitmentInput{
				CommitmentID: w.Node.Protocol.MainEngineInstance().Storage.Settings().LatestCommitment().Commitment().MustID(),
			},
		}),
		WithInputs(utxoledger.Outputs{input}),
		WithOutputs(iotago.Outputs[iotago.Output]{accountOutput}),
		WithSlotCreated(creationSlot),
		func(txBuilder *builder.TransactionBuilder) {
			txBuilder.AllotAllMana(creationSlot, implicitAccountID)
		},
	))
```

If you rewrite it like that, you can get rid of `createSignedTransactionWithAllotmentAndOptions` so there's less duplication.

Reply:
Nice, that is much cleaner